stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/internal/Channel.kt

```diff
@@ -52,24 +52,114 @@ internal fun Channel.updateLastMessage(message: Message) {
+ * Removes member from the [Channel.members] and aligns [Channel.memberCount].
  *
- * @param userId User id of the added/removed/updated member.
- * @param member Member object.
- * @param isUpdate True if a member is updated but not added/removed, false otherwise.
+ * @param currentUserId User id of the currently logged in user.
+ * @param memberUserId User id of the removed member.
  */
-internal fun Channel.updateMembers(userId: String, member: Member?, isUpdate: Boolean) {
-    if (member == null) {
-        members.firstOrNull { it.user.id == userId }?.also { foundMember ->
-            members = members - foundMember
-            memberCount -= 1
+internal fun Channel.removeMember(currentUserId: String?, memberUserId: String?): Channel {
+    memberUserId ?: return this
+    if (currentUserId == memberUserId) return this
+    members = members.filterNot { it.user.id == memberUserId }
+    memberCount -= 1
+    return this
+}
+
+/**
+ * Adds member to the [Channel.members] and aligns [Channel.memberCount].
+ *
+ * @param member Added member.
+ */
+internal fun Channel.addMember(member: Member?): Channel {
+    val memberUserId = member?.user?.id ?: return this
+    val foundMember = members.find { it.user.id == memberUserId }
+    if (foundMember != null) return this
+    members = members + member
+    memberCount += 1
+    return this
+}
+
+/**
+ * Updates member to the [Channel.members].
+ *
+ * @param member Updated member.
+ */
+internal fun Channel.updateMember(member: Member?): Channel {
+    val memberUserId = member?.user?.id ?: return this
+    members = members.map { iterableMember ->
+        when (iterableMember.user.id == memberUserId) {
+            true -> member
+            else -> iterableMember
         }
-    } else {
-        members = members + member
-        if (!isUpdate) {
-            memberCount += 1
+    }
+    return this
+}
+
+/**
+ * Updates member in the [Channel.members].
+ *
+ * @param memberUserId Updated member user id.
+ * @param banned Shows whether a user is banned or not in this channel.
+ */
+internal fun Channel.updateMember(memberUserId: String?, banned: Boolean): Channel {
+    members = members.map { member ->
+        member.apply {
+            if (this.user.id == memberUserId) {
+                this.banned = banned
+            }
         }
     }
+    return this
+}
+
+/**
+ * Sets [Channel.membership] to [member] if [currentUserId] equals to [member.user.id].
+ *
+ * @param currentUserId User id of the currently logged in user.
+ * @param member Added member.
+ */
+internal fun Channel.addMembership(currentUserId: String?, member: Member?): Channel {
+    if (member?.user?.id == currentUserId) {
+        membership = member
+    }
+    return this
+}
+
+/**
+ * Sets [Channel.membership] to [member] if [member.user.id] equals to [Channel.membership.user.id].
+ *
+ * @param member Updated member.
+ */
+internal fun Channel.updateMembership(member: Member?): Channel {
```

Comment:
What's the difference between this and `addMembership`? 🤔 
`Channel.membership` is only applicable for the current user, right? Maybe we can merge those two methods

Reply:
Inside the `Channel.updateMembership` we rely ing on `membership` existence , while in `addMembership` we assume that `membership` is most likely `null`.

Reply:
makes sence